sklearn/base.py

```diff
@@ -225,12 +225,27 @@ def get_params(self, deep=True):
             Parameter names mapped to their values.
         """
         out = dict()
-        for key in self._get_param_names():
```

Comment:
This is getting a bit adventurous of you! Propose this separately. It's not exclusive to this change.

Reply:
It somehow is, in the sense that it is quite essential for this PR, as otherwise just showing the repr of the new OneHotEncoder shows deprecation warnings. 

Did it happen before that keyword arguments were deprecated? How was it dealt with then?

I can certainly do it in a separate PR, but then that PR would be a blocker for this one IMO (which is not necessarily a problem, so fine for me to do that).